include/exadg/operators/inverse_mass_operator.h

```diff
@@ -60,14 +62,50 @@ class InverseMassOperator
     this->matrix_free = &matrix_free_in;
     dof_index         = dof_index_in;
     quad_index        = quad_index_in;
+
+    cellwise_inverse_mass_not_available = not(matrix_free->get_dof_handler(dof_index)
+                                                .get_triangulation()
+                                                .all_reference_cells_are_hyper_cube());
+
+    if(cellwise_inverse_mass_not_available)
+    {
+      initialize_inverse_mass_operator_with_block_jacobi();
+    }
+  }
+
+  void
+  initialize_inverse_mass_operator_with_block_jacobi()
+  {
+    // initialize mass operator
+    dealii::AffineConstraints<Number> const & constraint =
+      matrix_free->get_affine_constraints(dof_index);
+
+    MassOperatorData<dim> mass_operator_data;
+    mass_operator_data.dof_index  = dof_index;
+    mass_operator_data.quad_index = quad_index;
+
+    mass_operator.initialize(*matrix_free, constraint, mass_operator_data);
+
+    // build a BlockJacobiPreconditioner and use the vmult(dst,src) for applying the inverse mass
+    // operator on  source the vector
+    mass_preconditioner =
+      std::make_shared<BlockJacobiPreconditioner<MassOperator<dim, n_components, Number>>>(
+        mass_operator);
   }
 
   void
   apply(VectorType & dst, VectorType const & src) const
   {
     dst.zero_out_ghost_values();
 
-    matrix_free->cell_loop(&This::cell_loop, this, dst, src);
+    if(cellwise_inverse_mass_not_available)
+    {
+      mass_preconditioner->vmult(dst, src);
+    }
+    else
+    {
+      matrix_free->cell_loop(&This::cell_loop, this, dst, src);
+    }
```

Comment:
I think the variable name `cellwise_inverse_mass_not_available` is not optimal and hinders understanding the code. A cellwise inverse mass is available for all the branches here, otherwise we could not use a block Jacobi preconditioner to invert this operator.
The irritation might come from the naming in deal.II. Cellwise inverse mass actually means that the inverse mass can be applied in a matrix-free fashion using sum-factorization. The emphasis is probably not on "cellwise".

Reply:
I suggest to also rename `CellwiseInverseMass` into `MatrixfreeInverseMass` used in this class. The name was ok before the present PR, but is very irritating with the changes of this PR.

Reply:
I agree. Changed the name to `ExplicitMatrixfreeInverseMass`, to distinguish between matrix-free CG solver and the explicit matrix-free operator of dealii.